src/components/text-input/TextInput.props.ts

```diff
@@ -0,0 +1,98 @@
+import { ChangeEventHandler, FocusEventHandler, ReactNode, ReactNodeArray } from 'react';
+
+import { TIconGlyph } from '../../foundations/icon';
+import { TButtonWidth } from '../button';
+
+import { TTextInputSizeToken } from './TextInput.types';
+
+type PTextInput = {
+    /**
+     * The label for the text input
+     * */
+    label?: string;
+    /**
+     * The placeholder text for the text input
+     * */
+    placeholder?: string;
+    /**
+     * whether the text input has an error
+     * @default false
+     * */
+    hasError?: boolean;
+    /**
+     * the size of the text input
+     * @default 'md'
+     * */
+    size?: TTextInputSizeToken;
+    /**
+     * the width of the text input
+     * @default 'full'
+     * */
+    width?: TButtonWidth;
+    /**
+     * custom background color
+     * @default theme.background.default
+     * */
+    backgroundColor?: string;
+    /**
+     * the leading icon of the input, usually indicating search
+     * */
+    leadingIcon?: TIconGlyph;
+    /**
+     * the trailing icon of the input, usually indicating clear
+     * */
+    trailingIcon?: TIconGlyph;
```

Comment:
nit: IMO this should be sorted in this order:
1. mandatory properties
2. optional properties with defaults
3. optional properties without defaults
4. `className` and `children` (although I might add this to the `BaseProps` together with the `AriaProps` types)

WDYT about point 4 @nevyangelova ?